project-api/project-api-test/src/test/java/io/ballerina/projects/test/TestBuildProject.java

```diff
@@ -1531,6 +1531,63 @@ public void testAutoUpdateWithBuildFile() throws IOException {
         Assert.assertFalse(thirdBuildJson.isExpiredLastUpdateTime());
     }
 
+    @Test(description = "test auto updating dependencies with build file after removing Dependencies.toml")
+    public void testAutoUpdateWithBuildFileWithoutDepsToml() throws IOException {
+        Path projectPath = RESOURCE_DIRECTORY.resolve("myproject");
+        // Delete build file and Dependencies.toml file if already exists
+        Files.deleteIfExists(projectPath.resolve(TARGET_DIR_NAME).resolve(BUILD_FILE));
+        Files.deleteIfExists(projectPath.resolve(DEPENDENCIES_TOML));
+
+        // Set sticky false, to imitate the default build command behavior
+        BuildOptionsBuilder buildOptionsBuilder = new BuildOptionsBuilder();
+        buildOptionsBuilder.sticky(false);
+        BuildOptions buildOptions = buildOptionsBuilder.build();
+
+        // 1) Initialize the project instance
+        BuildProject project = null;
+        try {
+            project = TestUtils.loadBuildProject(projectPath, buildOptions);
+            project.save();
+        } catch (Exception e) {
+            Assert.fail(e.getMessage());
+        }
+        Assert.assertEquals(project.currentPackage().packageName().toString(), "myproject");
+        Path buildFile = project.sourceRoot().resolve(TARGET_DIR_NAME).resolve(BUILD_FILE);
+        Assert.assertTrue(buildFile.toFile().exists());
+        BuildJson initialBuildJson = readBuildJson(buildFile);
+        Assert.assertTrue(initialBuildJson.lastBuildTime() > 0);
+        Assert.assertTrue(initialBuildJson.lastUpdateTime() > 0);
+        Assert.assertFalse(initialBuildJson.isExpiredLastUpdateTime());
+        Assert.assertEquals(
+                readFileAsString(projectPath.resolve(RESOURCE_DIR_NAME).resolve("expectedDependencies.toml")),
+                readFileAsString(projectPath.resolve(DEPENDENCIES_TOML)));
+
+
+        // 2) Build project again with build file after removing Dependencies.toml
+        Files.deleteIfExists(projectPath.resolve(DEPENDENCIES_TOML));
+        Assert.assertTrue(projectPath.resolve(TARGET_DIR_NAME).resolve(BUILD_FILE).toFile().exists());
+        BuildProject projectSecondBuild = null;
+        try {
```

Comment:
Shall we extract project loading to a separate method to improve readability? Can be reused in any other tests as well. WDYT?

Reply:
Extracted a method and reused it.